src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Interop/MsQuicSafeHandle.cs

```diff
@@ -0,0 +1,45 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.InteropServices;
+using Microsoft.Quic;
+
+namespace System.Net.Quic.Implementations.MsQuic.Internal
+{
+    internal abstract class MsQuicSafeHandle : SafeHandle
+    {
+        private readonly Action<IntPtr> _releaseAction;
+        private readonly string _traceId;
+
+        public override bool IsInvalid => handle == IntPtr.Zero;
+
+        public unsafe QUIC_HANDLE* QuicHandle => (QUIC_HANDLE*)DangerousGetHandle();
+
+        protected unsafe MsQuicSafeHandle(QUIC_HANDLE* handle, Action<IntPtr> releaseAction, string prefix)
```

Comment:
Since the handle types are well-known, maybe just an `enum` for the type instead of a `string`?